library/common/engine.cc

```diff
@@ -32,23 +33,31 @@ envoy_status_t Engine::run(const std::string config, const std::string log_level
 envoy_status_t Engine::main(const std::string config, const std::string log_level) {
   // Using unique_ptr ensures main_common's lifespan is strictly scoped to this function.
   std::unique_ptr<EngineCommon> main_common;
+  const std::string name = "envoy";
+  const std::string config_flag = "--config-yaml";
+
+  size_t header_size = config_header.size();
```

Comment:
up to you about this vs using std:string and abseils concat utils. My paranoid stance is to go for the latter.

cc @mattklein123 in case he has a preference.

Reply:
I switched to MemBlockBuilder which is Envoy-provided and avoids raw memcpy.

Reply:
Switching to absl::StrCat since I think the docs make it sound trustworthy enough. :p